lib/ts/supertokens.ts

```diff
@@ -138,96 +137,15 @@ export default class SuperTokens {
         throw new Error("Initialisation not done. Did you forget to call the SuperTokens.init function?");
     }
 
-    // instance functions below......
-
-    middleware = () => {
-        return async (request: express.Request, response: express.Response, next: express.NextFunction) => {
-            let path = this.appInfo.apiGatewayPath.appendPath(
-                new NormalisedURLPath(request.originalUrl === undefined ? request.url : request.originalUrl)
-            );
-            let method: HTTPMethod = normaliseHttpMethod(request.method);
-
-            // if the prefix of the URL doesn't match the base path, we skip
-            if (!path.startsWith(this.appInfo.apiBasePath)) {
-                return next();
-            }
-
-            let requestRID = getRIDFromRequest(request);
-            if (requestRID !== undefined) {
-                let matchedRecipe: RecipeModule | undefined = undefined;
-
-                // we loop through all recipe modules to find the one with the matching rId
-                for (let i = 0; i < this.recipeModules.length; i++) {
-                    if (this.recipeModules[i].getRecipeId() === requestRID) {
-                        matchedRecipe = this.recipeModules[i];
-                        break;
-                    }
-                }
-
-                if (matchedRecipe === undefined) {
-                    // we could not find one, so we skip
-                    return next();
-                }
-
-                let id = matchedRecipe.returnAPIIdIfCanHandleRequest(path, method);
-                if (id === undefined) {
-                    // the matched recipe doesn't handle this path and http method
-                    return next();
-                }
-
-                // give task to the matched recipe
-                return await this.handleAPI(matchedRecipe, id, request, response, next, path, method);
-            } else {
-                // we loop through all recipe modules to find the one with the matching path and method
-                for (let i = 0; i < this.recipeModules.length; i++) {
-                    let id = this.recipeModules[i].returnAPIIdIfCanHandleRequest(path, method);
-                    if (id !== undefined) {
-                        return await this.handleAPI(this.recipeModules[i], id, request, response, next, path, method);
-                    }
-                }
-                return next();
-            }
-        };
-    };
-
     handleAPI = async (
```

Comment:
You don't actually need this function anymore...

Reply:
yeah correct